celesta.js

```diff
@@ -1296,10 +1298,16 @@
                 }
                 break;
             case KEYCODE_SPACE:
-                if (this._config.space_key_open) {
-                    this.open();
-                    // Don't prevent space keypress when already open (used for typed search or default browser action)
-                    if (!this._is_open) {
+                if (this._is_open) {
+                //if we not in the middle of typing, select hovered option
+                    if (this._typed == '' && this._config.space_key_select) {
```

Comment:
please consider making the following changes:
* add brackets around operator expression containing spaces (i.e.: around `this._typed == ''`)
* use strict equal operator (`===`)
* better check/compare `this._typed.length`, not `this._typed`
* check config value first, as boolean value check is faster, and the option is disabled by default